Makefile

```diff
@@ -7,7 +7,7 @@ build_dir := build-$(subst /,-,$(build_dir:/%=%))
 
 triplet := $(shell $(CXX) -dumpmachine)
 
-CXXFLAGS+=-Wno-attributes -Wno-unknown-pragmas -Wno-psabi
+CXXFLAGS+=-Wall -Wextra -Wpedantic -Wno-attributes -Wno-unknown-pragmas -Wno-psabi
```

Comment:
At some point, you might want to look at clang-tidy as well. I like some of their checks.

Reply:
I think, the signal to noise ration highly depends on your project and your general coding practices.

That said, in the long run, your code might get included (even embedded) in other projects that utilize clang-tidy. And they'll be thankful, if your code doesn't add to their noise. (I have to filter out various stuff from "included" projects in our builds).

Anyway: This was only a long term "Might consider". Not even close to a recommendation.

Reply:
No, you're right, of course. And most of these tools help more than they hurt. I added e.g. `-Wpedantic` because a user compiles with it and it produces noise. It's not helping anything, but I cleaned it up by adding `#pragma GCC diagnostic ignored "-Wpedantic"` where necessary. Ugh. I have to use a few non-standard things / extensions for SIMD. Kind of comes with the territory. Having my code in libstdc++ makes life easier in that regard. Then the warnings are silenced unless you ask for `-Wsystem-headers`...

I'm also using [Codacy](https://app.codacy.com/gh/mattkretz/vir-simd/dashboard). And it complained so much about `memcpy` being unchecked (which just isn't true - well there are preconditions on loads and stores of SIMD...) that I got tired of clicking "false positive" and disabled the rule altogether. Sometimes these checkers are simply too mechanical and don't understand enough context (certainly also because pre-conditions are not part of the code). And making the code more verbose to silence a warning on otherwise perfectly fine code is reducing overall quality of the code, IMHO.